src/responder/common/cache_req/cache_req.c

```diff
@@ -1312,8 +1312,10 @@ static void cache_req_input_parsed(struct tevent_req *subreq)
         return;
     }
 
-    state->domain_name = domain;
-    ret = cache_req_select_domains(req, domain,
+    if (state->domain_name == NULL) {
+        state->domain_name = domain;
+    }
+    ret = cache_req_select_domains(req, state->domain_name,
```

Comment:
Looks good to me.

But @pbrezina, please verify this specific patch.

Reply:
LGTM